### PR TITLE
add virtual memory page stats script

### PR DESCRIPTION
**Pull request by juergenthomann**
keeps track of:
- pages in
- pages out
- minor page faults
- major page faults

### CODE REVIEW

File: linux_vmstat.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/env python
+#
+# igcollect - Linux virtual memory stats
+#
+# Copyright (c) 2016, InnoGames GmbH
+#
+
+from __future__ import print_function
```
Comment (hasegeli) [inline linux_vmstat.py]: This is not necessary.

File: linux_vmstat.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+from socket import gethostname
+import time, sys
```
Comment (hasegeli) [inline linux_vmstat.py]: PEP 8 says they should go to separate lines.

File: linux_vmstat.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+# utility to read and parse a space delimited file (meminfo)
+def parse_split_file(filename):
+    try:
+        with open(filename, 'r') as f:
+            return [line.strip().split(None, 1) for line in f]
+    except:
```
Comment (hasegeli) [inline linux_vmstat.py]: It is always a bad idea to `expect` anything.  You would hide things like syntax errors, or Ctrl+C, if they happen to be inside this.  I don't thing we need any reason to expect anything in here.

File: linux_vmstat.py [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+vmstat = get_vmstat()
+timestamp = str(int(time.time()))
+hostname = gethostname().replace('.', '_')
+
+# Define desired fileds for ouput
+used_fields = ['pgfault', 'pgmajfault', 'pgpgin', 'pgpgout']
+
+template = "servers.{0}.system.virtualmemory.{1} {2} {3}"
+for field in used_fields:
+    print(template.format(hostname, field, vmstat[field], timestamp))
```
Comment (hasegeli) [inline linux_vmstat.py]: It is better style to put the rest inside `main()`.

File: linux_vmstat.py [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+def get_vmstat():
+    lines = parse_split_file('/proc/vmstat')
+    return dict((key, ( int(value.split()[0]))) for key, value in lines)
+
+vmstat = get_vmstat()
+timestamp = str(int(time.time()))
+hostname = gethostname().replace('.', '_')
+
+# Define desired fileds for ouput
+used_fields = ['pgfault', 'pgmajfault', 'pgpgin', 'pgpgout']
```
Comment (hasegeli) [inline linux_vmstat.py]: I would make this an argument.

File: linux_vmstat.py [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+def parse_split_file(filename):
+    try:
+        with open(filename, 'r') as f:
+            return [line.strip().split(None, 1) for line in f]
+    except:
+        sys.exit(1)
+
+def get_vmstat():
+    lines = parse_split_file('/proc/vmstat')
+    return dict((key, ( int(value.split()[0]))) for key, value in lines)
```
Comment (hasegeli) [inline linux_vmstat.py]: You can also use dict comprehension syntax: `{key: int(value.split()[0]) for key, value in lines}`

File: linux_vmstat.py [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+import time, sys
+
+# utility to read and parse a space delimited file (meminfo)
+def parse_split_file(filename):
+    try:
+        with open(filename, 'r') as f:
+            return [line.strip().split(None, 1) for line in f]
+    except:
+        sys.exit(1)
+
```
Comment (hasegeli) [inline linux_vmstat.py]: PEP 8 says two new lines between functions.

File: linux_vmstat.py (hunk 6 above)
Comment (TheFRedFox) [inline linux_vmstat.py]: 22:24: E201 whitespace after '('

File: linux_vmstat.py [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+#
+# Copyright (c) 2016, InnoGames GmbH
+#
+
+from __future__ import print_function
+from socket import gethostname
+import time, sys
+
+# utility to read and parse a space delimited file (meminfo)
+def parse_split_file(filename):
```
Comment (TheFRedFox) [inline linux_vmstat.py]: PEP8: 13:1: E302 expected 2 blank lines, found 1

File: linux_vmstat.py [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+            return [line.strip().split(None, 1) for line in f]
+    except:
+        sys.exit(1)
+
+def get_vmstat():
+    lines = parse_split_file('/proc/vmstat')
+    return dict((key, ( int(value.split()[0]))) for key, value in lines)
+
+vmstat = get_vmstat()
+timestamp = str(int(time.time()))
```
Comment (TheFRedFox) [inline linux_vmstat.py]: IMO timestamp should be an `int` and not a `string`. You should safely use `int` below in the `format` methods.

File: linux_vmstat.py [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+hostname = gethostname().replace('.', '_')
+
+# Define desired fileds for ouput
```
Comment (TheFRedFox) [inline linux_vmstat.py]: type "fileds" => "files" and "ouput" => "output"

File: linux_vmstat.py (hunk 3 above)
Comment (TheFRedFox) [inline linux_vmstat.py]: What about when one has no read access for the file. Will be raise an error then? Maybe we should just let it be raised.

Comment (hasegeli) [inline linux_vmstat.py]: Sure.  Python raise exceptions for all kinds of failures.

**Pull request merged**